src/control/destination/background/mod.rs

```diff
@@ -194,12 +198,20 @@ where
                     let config = &self.config;
                     let dsts = &mut self.dsts;
 
+                    // If the requested authority currently needs more
+                    // query capacity to query the destination service, go
+                    // ahead and try to free up capacity (by dropping any
+                    // inactive DestinationSets).
+                    if dsts.needs_query_for(&resolve.authority) {
+                        trace!("--> no query capacity, try retain_active...", );
+                        dsts.retain_active();
+                    };
+
                     match dsts.destinations.entry(resolve.authority) {
                         Entry::Occupied(mut occ) => {
-                            let set = occ.get_mut();
```

Comment:
This was necessary because of the borrow checker. `occ.get_mut()` creates a mutable borrow on `occ`, and because it's assigned to the variable `set`, that borrow lasts for the whole match arm. This was previously fine, but now we also need to get the `key` from the entry to pass to `query_destination_service_if_relevant`, and we can't call `occ.key()` while it's already borrowed by `get_mut()`:
```
error[E0502]: cannot borrow `occ` as immutable because it is also borrowed as mutable
   --> src/control/destination/background/mod.rs:233:41
    |
212 |                             let set = occ.get_mut();
    |                                       --- mutable borrow occurs here
...
233 |                                         occ.key(),
    |                                         ^^^ immutable borrow occurs here
...
241 |                         },
    |                         - mutable borrow ends here

error: aborting due to previous error
```

Reply:
An alternative to this would be something like this:
```rust
Entry::Occupied(mut occ) => {
    let authority = occ.key().clone();
    let set = occ.get_mut()
    // ...
```
However, _always_ cloning the authority regardless of whether or not we'll need to pass it to `query_destination_service_if_relevant` seems...less than ideal.

Reply:
cool, thanks